Gems/ROS2/Code/CMakeLists.txt

```
@@ -0,0 +1,193 @@
+    INCLUDE_DIRECTORIES
+        INTERFACE
+            Include
+)
+
+# Here add ROS2 target, it depends on the ROS2.Static
+ly_add_target(
+    NAME ROS2 ${PAL_TRAIT_MONOLITHIC_DRIVEN_MODULE_TYPE}
+    NAMESPACE Gem
+    FILES_CMAKE
+        ros2_shared_files.cmake
+    INCLUDE_DIRECTORIES
+        PUBLIC
+            Include
+        PRIVATE
+            Source
+    BUILD_DEPENDENCIES
+        PRIVATE
+            Gem::ROS2.Static
+            Gem::Atom_Feature_Common.Static
+)
+
+# By default, we will specify that the above target ROS2 would be used by
+# Client and Server type targets when this gem is enabled.  If you don't want it
+# active in Clients or Servers by default, delete one of both of the following lines:
+ly_create_alias(NAME ROS2.Clients NAMESPACE Gem TARGETS Gem::ROS2)
+ly_create_alias(NAME ROS2.Servers NAMESPACE Gem TARGETS Gem::ROS2)
+
+# If we are on a host platform, we want to add the host tools targets like the ROS2.Editor target which
+# will also depend on ROS2.Static
+if(PAL_TRAIT_BUILD_HOST_TOOLS)
+    ly_add_target(
+        NAME ROS2.Editor.Static STATIC
+        NAMESPACE Gem
+        FILES_CMAKE
+            ros2_editor_files.cmake
+        PLATFORM_INCLUDE_FILES
+            ${CMAKE_CURRENT_LIST_DIR}/Platform/Common/${PAL_TRAIT_COMPILER_ID}/ros2_static_editor_${PAL_TRAIT_COMPILER_ID_LOWERCASE}.cmake
+        INCLUDE_DIRECTORIES
+            PRIVATE
+                Source
+            PUBLIC
+                Include
+        COMPILE_DEFINITIONS
+            PRIVATE
+                ROS2_EDITOR
+        BUILD_DEPENDENCIES
+            PUBLIC
+                AZ::AzToolsFramework
+                Gem::AtomLyIntegration_CommonFeatures.Editor.Static
+                Gem::LmbrCentral.API
+                Gem::PhysX.Editor.Static
+                Gem::ROS2.Static
+    )
+
+    ly_add_target(
+        NAME ROS2.Editor GEM_MODULE
+        NAMESPACE Gem
+        AUTOMOC
+        AUTORCC
+        FILES_CMAKE
+            ros2_editor_files.cmake
+            ros2_editor_shared_files.cmake
+        PLATFORM_INCLUDE_FILES
+            ${CMAKE_CURRENT_LIST_DIR}/Platform/Common/${PAL_TRAIT_COMPILER_ID}/ros2_static_editor_${PAL_TRAIT_COMPILER_ID_LOWERCASE}.cmake
+        COMPILE_DEFINITIONS
+            PRIVATE
+                ROS2_EDITOR
+        INCLUDE_DIRECTORIES
+            PRIVATE
+                Source
+            PUBLIC
+                Include
+        BUILD_DEPENDENCIES
+            PUBLIC
+                Gem::ROS2.Editor.Static
+                Gem::Atom_Feature_Common.Static
+    )
+
+    # By default, we will specify that the above target ROS2 would be used by
+    # Tool and Builder type targets when this gem is enabled.  If you don't want it
+    # active in Tools or Builders by default, delete one of both of the following lines:
+    ly_create_alias(NAME ROS2.Tools    NAMESPACE Gem TARGETS Gem::ROS2.Editor)
+    ly_create_alias(NAME ROS2.Builders NAMESPACE Gem TARGETS Gem::ROS2.Editor)
```

repeat: We can use the `${gem_name}` variable as a prefix for the CMake targets for easier renaming.
```suggestion
    ly_create_alias(NAME ${gem_name}.Tools    NAMESPACE Gem TARGETS Gem::${gem_name}.Editor)
    ly_create_alias(NAME ${gem_name}.Builders NAMESPACE Gem TARGETS Gem::${gem_name}.Editor)
```